### DISCUSSION TITLE
Why does Aptos Compiler allow custom structs in Entry Functions despite SDK limitations?

### DISCUSSION THREAD

**Discussion by lumos42** [aptos-labs/aptos-developer-discussions#409] [Questions]
While I've seen at a bunch of places that entry functions support only native types, aptos compiler allows for having custom structs in entry functions too. Is this a missed check in the compiler?

for example, i'm able to create a function like this:
```rust
public entry fun create_pool<T: drop + store + copy>(
        sender: &signer,
        name: String,
        token: Object<Metadata>,
        deposit_cap: u128,
        rate_model: u8,
        rate_model_data: T
)
```

It lets me compile, and deploy this function, and also use it in move scripts, but there's no support in aptos-sdk to call this function. So it this allowed or not?

I've already seen a few discussions/comments that custom structs are not supported. My question is - why does the compiler allow for it?
1. https://github.com/aptos-labs/aptos-developer-discussions/discussions/275
2. https://github.com/aptos-labs/aptos-developer-discussions/discussions/176#discussioncomment-9332773

**Comment by kshitijc1506** [comment]
You're referring to passing T as a param? 

From what I can tell this compiles but will give a runtime error if T is a custom struct. Probably, it's the case that custom types can compile because the compiler doesn't enforce runtime execution checks around external transaction signing, serialization, and deserialization for complex custom types.

**Reply by lumos42** [reply]
@kshitjee I tried calling the create_pool function from a move script, and it worked. You can see the transaction here:

https://explorer.aptoslabs.com/txn/0x21353a10fd9d8c235ed92228afa877e7fd5d27d531faf5e2069ccc3a8ffb2aef?network=testnet

**Comment by kshitijc1506** [comment]
Also, in the future, could you pls change the categories of questions you're posting to "Questions"? Really appreciate, thx 🙏
